server/datastore/mysql/software.go

```diff
@@ -76,16 +80,152 @@ func (d *Datastore) SaveHostSoftware(host *fleet.Host) error {
 	return nil
 }
 
-func (d *Datastore) LoadHostSoftware(host *fleet.Host) error {
-	host.HostSoftware = fleet.HostSoftware{Modified: false}
+func nothingChanged(current []fleet.Software, incoming []fleet.Software) bool {
+	if len(current) != len(incoming) {
+		return false
+	}
+
+	currentBitmap := make(map[string]bool)
+	for _, s := range current {
+		currentBitmap[softwareToUniqueString(s)] = false
```

Comment:
I find it a bit easier to follow when `true` is used for sets (even though it doesn't change the behavior)
```suggestion
		currentBitmap[softwareToUniqueString(s)] = true
```